### PR TITLE
[#6169] Display .eth or other domain rather than IPFS redirect

### LINKED ISSUE #6169: Display .eth or other domain rather than IPFS redirect
### User Story

As a user, I want to receive permission requests from a specific URL—not `ipfs.infura`—so that I am aware of who is requesting my data.

### Description

*Type*: Feature

*Summary*: Rather than displaying a hash for IPFS-hosted DApps in the browser, keep the human readable format of the redirected domain. 

#### Expected behavior
Instead of `https://ipfs.infura.io/ipfs/QmYhTGg88WvJh9K86m3JYow3QxdXMkgk8stkTDqy87joZn/`, for example, I would see `statusnames.eth` in the URL bar. Any requests from this DApp would come from `statusnames` (if not a default DApp in Status). 

#### Actual behavior
Currently, the browser displays the IPFS hash a domain is redirected to, and this in turn is used when the DApp requests user permissions for something. 

### Reproduction
Navigate to `vote.status.im` in any browser.

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #6169

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

File: src/status_im/browser/core.cljs [hunk 1]
```diff
@@ -61,9 +73,35 @@
   (let [history-host (http/url-host (try (nth history history-index) (catch js/Error _)))]
     (assoc browser :unsafe? (dependencies/phishing-detect history-host))))
 
+(defn ens? [host]
+  (and (string? host)
+       (string/ends-with? host ".eth")))
```
Comment (jeluard) [inline src/status_im/browser/core.cljs]: In the future others domain will be available so we will need something more robust.
For now can we make this a utility and share with others part of the codebase? i.e. pretty sure chat already has this somewhere

File: src/status_im/browser/core.cljs [hunk 2]
```diff
@@ -91,17 +128,19 @@
...
+  [{db :db :as cofx} browser url loading?]
   (when-not loading?
     (let [history-index (:history-index browser)
           history       (:history browser)
-          history-url   (get-current-url browser)]
-      (when (not= history-url url)
-        (let [slash?      (= url (str history-url "/"))
+          current-url   (get-in db [:browser/options :url])]
+      (when (and (not= current-url url) (not= (str current-url "/") url))
+        (let [slash?      (= url (str current-url "/"))
```
Comment (jeluard) [inline src/status_im/browser/core.cljs:147]: Confusing. Maybe simplify?

File: src/status_im/browser/core.cljs [hunk 3]
```diff
@@ -113,40 +152,18 @@
...
-                                                      chain)
-                                       :ens-name host
-                                       :cb       resolve-ens-multihash-callback}})))
+              {:db (-> (update db :browser/options
+                               assoc
+                               :url (str "https://ipfs.infura.io/ipfs/"
+                                         hash
+                                         (subs current-url (+ (.indexOf current-url host) (count host))))
+                               :resolving? false)
+                       (assoc-in [:browser/options :resolved-ens host] (str "ipfs.infura.io/ipfs/" hash)))})))
```
Comment (jeluard) [inline src/status_im/browser/core.cljs]: Twice same URL but one w/o https?

File: src/status_im/browser/core.cljs [hunk 4]
```diff
@@ -160,11 +177,14 @@
...
-  (when (not= "about:blank" url)
-    (let [host (http/url-host url)]
-      (fx/merge cofx
-                (resolve-ens-multihash host loading? error?)
-                (update-browser-history browser url loading?)))))
+  (let [options (get-in cofx [:db :browser/options])
+        current-url (:url options)]
+    (when (and (not= "about:blank" url) (not= current-url url) (not= (str current-url "/") url))
+      (let [resolved-ens (first (filter #(not= (.indexOf url (second %)) -1) (:resolved-ens options)))
+            resolved-url (if resolved-ens (string/replace url (second resolved-ens) (first resolved-ens)) url)] (println "update-browser-on-nav-change" loading? error?)
```
Comment (jeluard) [inline src/status_im/browser/core.cljs]: println

**Review by jeluard — CHANGES_REQUESTED** [review]

**Comment by flexsurfer** [comment]
thanks @jeluard , fixed

**Review by jeluard — APPROVED** [review]

**Comment by lukaszfryc** [comment]
@flexsurfer @rachelhamlin 

1) Could you provide some ens domains (that lead to ipfs) that I could test on? I tried `statusnames.eth` but it does not load (no url shown either):


 
2) Also, should I be redirected to `statusnames.eth` if I typed `https://ipfs.infura.io/ipfs/QmYhTGg88WvJh9K86m3JYow3QxdXMkgk8stkTDqy87joZn/` in the browser?

**Comment by flexsurfer** [comment]
@lukaszfryc 
1) names.statusnet.eth
2) no , we don't support reverse resolution

**Comment by flexsurfer** [comment]
@lukaszfryc made a fix for names which can't be resolved, show error

**Pull request merged**